test/gtest/uct/test_p2p_am.cc

```diff
@@ -339,4 +390,46 @@ UCS_TEST_P(uct_p2p_am_test, am_zcopy) {
 }
 
 
+const unsigned uct_p2p_am_misc::RX_MAX_BUFS = 1024; /* due to hard coded 'grow'
+                                                       parameter in uct_ib_iface_recv_mpool_init */
+const unsigned uct_p2p_am_misc::RX_QUEUE_LEN = 64;
+
+UCS_TEST_P(uct_p2p_am_misc, no_rx_buffs) {
+
+    mapped_buffer sendbuf(10 * sizeof(uint64_t), SEED1, sender());
+    mapped_buffer recvbuf(0, 0, sender()); /* dummy */
+    ucs_status_t status;
+    uint64_t i;
+
+    if (m_rx_buf_limit_failed) {
+        UCS_TEST_SKIP_R("Current transport doesn't have rx memory pool");
+    }
+    check_caps(UCT_IFACE_FLAG_AM_SHORT);
+
+    /* set a callback for the uct to invoke for receiving the data */
+    status = uct_iface_set_am_handler(receiver().iface(), AM_ID, am_handler, (void*)this, UCT_AM_CB_FLAG_SYNC);
+    ASSERT_UCS_OK(status);
+
+    /* send many messages and progress the receiver. the receiver will keep getting
+     * UCS_INPROGRESS from the recv-handler and will keep consuming its rx memory pool.
+     * the goal is to make the reciever's rx memory pool run out */
+    set_keep_data(true);
+    for (i = 0; i < (5 * RX_MAX_BUFS); i++) {
+        status = am_short(sender_ep(), sendbuf, recvbuf);
+        progress();
+    }
+    /* check that now the sender cannot send for a while as it was stopped by the receiver */
+    EXPECT_EQ(UCS_ERR_NO_RESOURCE, send_with_timeout(sender_ep(), sendbuf, recvbuf, 1, 0));
```

Comment:
calling progress made rc send the message and fail the test. 
i can increase the iterations in the 'for' above to (3000 * RX_MAX_BUFS) so that rc would pass (with calling progress)